python/cudf/cudf/core/dataframe.py

```diff
@@ -1992,6 +1992,250 @@ def _make_operands_and_index_for_binop(
+            of tuples or dicts, or DataFrame.
+        DataFrame : DataFrame object creation using constructor.
+        DataFrame.to_dict : Convert the DataFrame to a dictionary.
+
+        Examples
+        --------
+        By default the keys of the dict become the DataFrame columns:
+
+        >>> import cudf
+        >>> data = {'col_1': [3, 2, 1, 0], 'col_2': ['a', 'b', 'c', 'd']}
+        >>> cudf.DataFrame.from_dict(data)
+           col_1 col_2
+        0      3     a
+        1      2     b
+        2      1     c
+        3      0     d
+
+        Specify ``orient='index'`` to create the DataFrame using dictionary
+        keys as rows:
+
+        >>> data = {'row_1': [3, 2, 1, 0], 'row_2': [10, 11, 12, 13]}
+        >>> cudf.DataFrame.from_dict(data, orient='index')
+                0   1   2   3
+        row_1   3   2   1   0
+        row_2  10  11  12  13
+
+        When using the 'index' orientation, the column names can be
+        specified manually:
+
+        >>> cudf.DataFrame.from_dict(data, orient='index',
+        ...                          columns=['A', 'B', 'C', 'D'])
+                A   B   C   D
+        row_1   3   2   1   0
+        row_2  10  11  12  13
+
+        Specify ``orient='tight'`` to create the DataFrame using a 'tight'
+        format:
+
+        >>> data = {'index': [('a', 'b'), ('a', 'c')],
+        ...         'columns': [('x', 1), ('y', 2)],
+        ...         'data': [[1, 3], [2, 4]],
+        ...         'index_names': ['n1', 'n2'],
+        ...         'column_names': ['z1', 'z2']}
+        >>> cudf.DataFrame.from_dict(data, orient='tight')
+        z1     x  y
+        z2     1  2
+        n1 n2
+        a  b   1  3
+           c   2  4
+        """  # noqa: E501
+        index = None
+        orient = orient.lower()
+        if orient == "index":
+
+            if len(data) > 0:
+                data_list = list(data.values())
+                if isinstance(data_list[0], (pd.Series, Series, dict)):
+                    if isinstance(data_list[0], Series):
+                        data = {key: val.to_pandas() for key, val in data}
+                    data = _from_nested_dict(data)
+                else:
+                    index = list(data.keys())
+                    data = data_list
+            df = pd.DataFrame(data=data, index=index, columns=columns)
+            df = cudf.from_pandas(df)
+            if dtype:
+                return df.astype(dtype)
+            return df
+        elif orient == "columns" or orient == "tight":
+            if columns is not None:
+                raise ValueError(
+                    f"cannot use columns parameter with orient='{orient}'"
+                )
+        else:
+            raise ValueError(
+                f"Expected 'index', 'columns' or 'tight' for orient "
+                f"parameter. Got '{orient}' instead"
+            )
+
+        if orient != "tight":
+            return cls(data, index=index, columns=columns, dtype=dtype)
```

Comment:
`columns` must be `None` here?

Reply:
Why do you think so? This is for `orient != 'tight'`. `columns` is fetched from `data` when `orient=='tight'`

Reply:
orient can be "columns", "index", or "tight"; if `orient == "index"` we have already returned by this point. Then on line 2103 you check if `orient == "columns" or orient == "tight"` (must succeed) and then on line 2104 if `columns is not None` we raise a ValueError. So at this point `orient != "tight"` means `orient == "columns"` and columns must have been `None` otherwise we would have bailed out with a `ValueError` on line 2105.

Reply:
Oh yes, you are right. Removed passing of columns.